lib/components/src/syntaxhighlighter/syntaxhighlighter.tsx

```diff
@@ -91,6 +91,31 @@ export interface SyntaxHighlighterState {
 
 type ReactSyntaxHighlighterProps = React.ComponentProps<typeof ReactSyntaxHighlighter>;
 
+const formatter = memoize(2)((code: string) => {
```

Comment:
This seems pretty haphazard to me. Can you factor this into a separate file and provide test cases?

Reply:
good call